dart/lib/src/protocol/sentry_device.dart

```diff
@@ -178,7 +178,9 @@ class SentryDevice {
         model: data['model'],
         modelId: data['model_id'],
         arch: data['arch'],
-        batteryLevel: data['battery_level'],
+        batteryLevel:
+            (data['battery_level'] is num ? data['battery_level'] as num : null)
+                ?.toDouble(),
```

Comment:
I miss the nice `as?` type of conversion that exists in other languages, so I don't have to check if the type `is` first.
We don't have that in Dart, right?

Reply:
You can do `data['battery_level'] as num?` but that will fail if you cast a string for example. If you cast null it works.

So, I guess the following should do it, since batter_level should always be a number.
```dart
(data['battery_level'] as num?)?.toDouble()
```
If it wasn't a number, there's another error somewhere else.

Reply:
I did with `num?` first but then it fails with a string, for example, "1" instead of 1, so I preferred to go safer checking if it's a `num` first.